main.go

```diff
@@ -1,5 +1,67 @@
 package main
 
+import (
+	"context"
+	"log"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/config"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/forman"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/reconciler"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+
+	"go.uber.org/zap"
+)
 
 func main() {
+	// parse configuration from yaml file
+	conf, err := config.GetConfig("config/config.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	logger, err := zap.NewProduction()
+	if conf.General.Development {
+		logger, err = zap.NewDevelopment()
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+	ctx := context.Background()
+	dbInstance, err := db.NewPGInstance(ctx, logger, &conf.DB)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	if err := dbInstance.Ping(ctx); err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	reqChan := forman.GoForman(logger, conf.Worker.Count, reconciler.Reconcile)
+
+	pgConn, err := dbInstance.SetupNotification(ctx, reqChan)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	defer pgConn.Close(ctx)
+
+	providers, err := dbInstance.ListProviderClusters(ctx)
```

Comment:
- is this a safety net against reboot? if yes, better to read the rows sequentially?